interface.go

```diff
@@ -123,21 +123,17 @@ func (c *Cache[T]) GetIndirect(key string, linkResolver func(*T) string) (*T, er
 
 // SetIndirect sets cache key including intermediary links
 func (c *Cache[T]) SetIndirect(key string, value *T, linkResolver func(*T) string, linkGenerator func(*T) *T) error {
-	c.computeLocks.Lock(key)
-	defer c.computeLocks.Unlock(key)
+	mutex := c.computeLocks.Lock(key)
+	defer c.computeLocks.Unlock(key, mutex)
 	if linkGenerator != nil && linkResolver != nil {
 		if linkValue := linkGenerator(value); linkValue != nil {
+			if err := c.setNoLock(key, linkValue); err != nil {
```

Comment:
This actually modifies the behavior, if key == link, it should Set(link [==key], value). In your new code you just Set(key, link), which is exactly the opposite of what it did before. Not sure it matters (if link == key actually happens in real life), but it well may.

Reply:
Thanks for being vigilant. Solved